app/src/Controllers/SearchController.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+
+namespace SilverStripe\Bambusa\Controllers;
+
+use SilverStripe\Bambusa\Search\PageIndex;
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\FullTextSearch\Search\Queries\SearchQuery;
+use PageController;
+use SilverStripe\View\ArrayData;
+use TractorCow\Fluent\Extension\FluentExtension;
+
+class SearchController extends PageController
+{
+
+    /**
+     * Map Macrons to non-macrons, otherwise they're stripped
+     * out of the query and replaced by a space
+     * @var array
+     */
+    private static $macrons = [
+        'ā' => 'a',
+        'ē' => 'e',
+        'ī' => 'i',
+        'ō' => 'o',
+        'ū' => 'u',
+        'Ā' => 'A',
+        'Ē' => 'E',
+        'Ī' => 'I',
+        'Ō' => 'O',
+        'Ū' => 'U'
+    ];
+
+
+    /**
+     * Because this is a standalone controller and not affected by the
+     * fluent routing, we have to ensure that the language selector
+     * generates ?l=$locale links. All other links on the page are rendered
+     * with the sub URL, so this parameter  doesn't get forwarded to subsequent
+     * page views.
+     *
+     * @param null $locale
+     * @return ArrayData
+     */
+    public function LocaleInformation($locale = null)
+    {
+        $data = Injector::inst()->get(FluentExtension::class)
+            ->LocaleInformation($locale);
+
+        $query = http_build_query([
+            'l' => $locale,
+            'q' => $this->getRequest()->getVar('q')
+        ]);
+
+        $link = $data->Link . '?' . $query;
+
+        $data->Link = $link;
```

Comment:
Can you check if this is XSS?

Reply:
Checked a very simple one, seems fine

![image](https://user-images.githubusercontent.com/111025/62440721-757b1c00-b7a5-11e9-88ee-8b7a379c08e2.png)